src/pages/strategies/forms/strategiesForm.js

```diff
@@ -0,0 +1,158 @@
+// ** MUI Imports
+import { Grid } from '@mui/material'
+import { DataSets } from 'src/resources/DataSets'
+
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+
+// ** Custom Imports
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import FormShell from 'src/components/Shared/FormShell'
+import { useFormik } from 'formik'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useContext, useEffect, useState, useRef } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
+import { useInvalidate } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+
+const StrategiesForm = ({ labels, editMode, maxAccess, setEditMode, setStore, store, onUpdateFormik }) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+  const { recordId } = store
+
+  const invalidate = useInvalidate({
+    endpointId: DocumentReleaseRepository.Strategy.qry
+  })
+
+  const [initialValues, setInitialData] = useState({
+    recordId: null,
+    name: '',
+
+    type: '',
+    groupId: ''
+  })
+
+  const strategiesFormik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues,
+    validationSchema: yup.object({
+      name: yup.string().required(),
+      groupId: yup.string().required(),
+      type: yup.string().required()
+    }),
+    onSubmit: values => {
+      postGroups(values)
+    }
+  })
+
+  const prevValuesRef = useRef()
+
+  useEffect(() => {
```

Comment:
Why do we need this? I don't like the prevValues, newValues comparison on this form.